citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -2223,12 +2244,36 @@ public String buildMainPanelContext(VelocityPortlet portlet, Context context, Ru
 			case VIEW:
 				template = buildViewPanelContext(portlet, context, rundata, state);
 				break;
+			case EDIT_ATTACHMENT:
+				template = buildEditAttachmentPanelContext(portlet,context,rundata,state);
 		}
 
 		return template;
 
 	}	// buildMainPanelContext
 
+	private String buildEditAttachmentPanelContext(VelocityPortlet portlet, Context context, RunData data, SessionState state) {
+		logger.debug("buildEditAttachmentPanelContext()");
+		ParameterParser params = data.getParameters();
+		state.removeAttribute("sakaiproject.filepicker.addAttachments");
+		try {
+			CitationCollection collection = getCitationCollection(state, true);
+			Citation citation = extractCitationFromRunData(data);
+			if(citation!=null) {
+				state.setAttribute(CitationHelper.CITATION_EDIT_ITEM, citation);
+				state.setAttribute(CitationHelper.CITATION_EDIT_ID,citation.getId());
+				//collection.add(citation);
+				state.setAttribute(STATE_CITATION_COLLECTION, collection);
+				buildEditPanelContext(portlet, context, data, state);
+			}
+
+		}catch(Exception e){
+
```

Comment:
Need some error handling here, if it's expected that the user can trigger it just display to the user, if it's unexpected log it and display something to the user.